### PR TITLE
feature: Added rrd step conversion script

### PR DESCRIPTION
**Pull request by laf**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

Works for me :)

### CODE REVIEW

**Comment by wriedel** [comment]
Hi Laf, it's a little bit cumbersome to convert 90 devices that way but it's being worth every minute to get this 1 minute high resolution graphs ;-) Script worked like a charm and I hope we can get a wildcard option in the near future.

**Comment by laf** [comment]
Added -h all.

Don't see the point in adding a wildcard.

File: scripts/rrdstep.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+$rrdtool   = $config['rrdtool'];
+$tmp_path  = $config['temp_dir'];
+
+if ($hostname === 'all') {
+    $hostname = '*';
+}
+$files = glob($rrd_path . '/' . $hostname . '/*.rrd');
+
+$run = readline("Are you sure you want to run this command [N/y]: ");
+if ($run != 'y' || $run == 'Y') {
```
Comment (murrant) [inline scripts/rrdstep.php]: Looks like this should be: 
`+if ($run != 'y' || $run != 'Y')`

File: scripts/rrdstep.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    exit;
+}
+
+foreach ($files as $file) {
+    $random = $tmp_path.'/'.mt_rand() . '.xml';
+    $tmp = explode('/', $file);
+    $rrd_file = array_pop($tmp);
+    echo "Converting $file: ";
+    $command = "$rrdtool dump $file > $random;
+        sed -i 's/<step>300/<step>$step/' $random;
```
Comment (murrant) [inline scripts/rrdstep.php]: Perhaps use && in between these commands.  That way if one fails the rest are not executed.

**Comment by murrant** [comment]
@laf I know you mentioned this elsewhere, but why can't you use rrdtune?

**Comment by laf** [comment]
Because it doesn't work, try it. If you tune a file and then dump it you will see the step will remain the same.

I'm unsure if this is all versions but (iirc) I tested on 1.4.7 and 1.6.x

**Comment by nrm21** [comment]
Hi,

This script works great!  However my graphs only update every 5 mins and they update with 5 new values.  On further inspection it appears the code is doing some sort of:
what is the current unix timestamp? ->
ok now lets round to the nearest 5 min step of that timestamp ->
we'll use that in the URL request for 'from=' and 'to=' ->
now we generate the graph using those values

...or it appears something like that bawsd on studying web behavior.

Maybe the graph portion of code could be changed to round to the nearest minute of the current timestamp instead of 5 min, once the script is applied?

**Comment by nrm21** [comment]
Found it... in ./includes/definitions.inc.php

Change:     $config['time']['now']     -= ($config['time']['now'] % 300);
from mod 300 to mod 60

and voila!  Graphs now show by the min.  :)

**Comment by laf** [comment]
Thanks but this has nothing to do with this script.

**Comment by nrm21** [comment]
Ok maybe it doesn't but.  I assume that others like me might want to not just poll every min but also see an up to the min graph of their bandwidth data.  In that case, it might be necessary to do that to the default code to achieve it.

**Comment by laf** [comment]
Feature requests need to be posted on https://community.librenms.org/c/feature-requests

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/1-minute-polling-and-graphs/787/9

**Pull request merged**